test/Microsoft.ML.TestFramework/BaseTestBaseline.cs

```diff
@@ -590,10 +550,22 @@ protected bool CheckOutputIsSuffix(string basePath, string outPath, int skip = 0
             return true;
         }
 
-        public bool CompareNumbersWithTolerance(double expected, double actual, int? iterationOnCollection = null, int digitsOfPrecision = DigitsOfPrecision)
+        public bool CompareNumbersAndLogErrors(double expected, double actual,
+            int? iterationOnCollection = null, int digitsOfPrecision = DigitsOfPrecision)
+        {
+            var compareResult = CompareNumbers(expected, actual, iterationOnCollection, digitsOfPrecision);
+
+            if (!compareResult.Item1)
+                Fail(compareResult.Item2);
+
+            return compareResult.Item1;
+        }
+
+        public (bool, string) CompareNumbers(double expected, double actual, 
+            int? iterationOnCollection = null, int digitsOfPrecision = DigitsOfPrecision, bool allowMismatch = false)
         {
```

Comment:
Sorry for changing my mind on this. As I read through your current implementation I see that it is resulting in a lot of changes. I am now thinking it may be better to add a default parameter instead. Instead of the previous allowMismatch param you added, would it be better to add a logFailure parameter that defaults to true and retain the same function name? That will eliminate the changes in the rest of the code and the only changes should be in places where we are comparing unequal numbers.